### ISSUE TITLE
Comprehensive report generation fails due to long response strings

### ISSUE THREAD

**Issue by muloem** [akvo/akvo-flow#637]
While generating a comprehensive report the following exception is observed

``` java
java.lang.IllegalArgumentException: The workbook already contains a sheet of this name
    at org.apache.poi.xssf.usermodel.XSSFWorkbook.createSheet(XSSFWorkbook.java:619)
    at org.apache.poi.xssf.streaming.SXSSFWorkbook.createSheet(SXSSFWorkbook.java:585)
    at org.waterforpeople.mapping.dataexport.GraphicalSurveySummaryExporter.writeSummaryReport(GraphicalSurveySummaryExporter.java:681)
    at org.waterforpeople.mapping.dataexport.GraphicalSurveySummaryExporter.export(GraphicalSurveySummaryExporter.java:357)
    at akvo.flow_services.exporter$export_report.invoke(exporter.clj:45)
    at akvo.flow_services.scheduler.ExportJob.execute(scheduler.clj:43)
    at org.quartz.core.JobRunShell.run(JobRunShell.java:213)
    at org.quartz.simpl.SimpleThreadPool$WorkerThread.run(SimpleThreadPool.java:557)
```

The issue was reported by @andrewmolo 
http://flowhelp.akvo.org/discussions/problems/1377-error-generating-comprehensive-report

This exception is a result of having strings longer than 31 characters the generated Excel sheets.  The library we use to create these sheets (https://github.com/akvo/akvo-flow/blob/develop/GAE/war/poi-3.8-20120326.jar) truncates a sheet name to 31 characters. 

While generating the comprehensive report, separate summary sheets are generated for responses to questions whose text is _District_, _County_ and _Sub County_. The names of these summary sheets are built as a rollup (pipe separated concatenation) of the `QuestionOption` values e.g. `DistrictOne|CountyThree|SubcountyFive`.  For cases where this rollup is longer than 31 characters, the exception is thrown.

https://github.com/akvo/akvo-flow/blob/release/1.6.17.1/GAE/src/org/waterforpeople/mapping/dataexport/GraphicalSurveySummaryExporter.java#L670-688

To reproduce:
1. Create a sample survey
2. Create a question group and add three option questions one titled _District_ another _County_ and another _Sub County_
3. Create dependencies in the hierarchy, i.e, a County dependent on a district and Sub County dependent on a county
4. For each of the questions ensure that the options are made up of very long string names greater than 15 characters
5. Fill out some sample surveys
6. Try to generate a comprehensive report

**Comment by muloem** [comment]
This is the same as  #1054.  Long responses results in failure to create a worksheet in the comprehensive reports.  The problem has been observed with cascade responses and option question responses

**Comment by janagombitova** [comment]
I think WHH is having the same problem: 

https://flowhelp.reamaze.com/admin/conversations/downloading-graphical-data WHH

**Comment by stellanl** [comment]
So, to make unique names that fit longer than 31 chars, something has to go. Some approaches to what to try before resorting to numbers or random strings:
* "..." followed by the last 28 chars
* The first 14 + "..." + last 14

**Comment by stellanl** [comment]
This is connected to #1968, where we tried to remove JSON from sheet names and #2206 where JSON is causing an infinite loop.

**Comment by stellanl** [comment]
Finally resolved in #2206
